src/ScriptEngine/ScriptEngine.cs

```diff
@@ -76,22 +76,39 @@ void LoadDLL(string path, GameObject obj)
                     dll.Write(ms);
                     var ass = Assembly.Load(ms.ToArray());
 
-                    foreach(Type type in ass.GetTypes())
+                    try
                     {
-                        if(typeof(BaseUnityPlugin).IsAssignableFrom(type))
+                        foreach (Type type in ass.GetTypes())
                         {
-                            var metadata = MetadataHelper.GetMetadata(type);
-                            if(metadata != null)
+                            if (typeof(BaseUnityPlugin).IsAssignableFrom(type))
                             {
-                                var typeDefinition = dll.MainModule.Types.First(x => x.FullName == type.FullName);
-                                var typeInfo = Chainloader.ToPluginInfo(typeDefinition);
-                                Chainloader.PluginInfos[metadata.GUID] = typeInfo;
-
-                                Logger.Log(LogLevel.Info, $"Reloading {metadata.GUID}");
-                                StartCoroutine(DelayAction(() => obj.AddComponent(type))); 
+                                var metadata = MetadataHelper.GetMetadata(type);
+                                if (metadata != null)
+                                {
+                                    var typeDefinition = dll.MainModule.Types.First(x => x.FullName == type.FullName);
+                                    var typeInfo = Chainloader.ToPluginInfo(typeDefinition);
+                                    Chainloader.PluginInfos[metadata.GUID] = typeInfo;
+
+                                    Logger.Log(LogLevel.Info, $"Reloading {metadata.GUID}");
+                                    StartCoroutine(DelayAction(() => obj.AddComponent(type)));
+                                }
                             }
                         }
                     }
+                    catch (ReflectionTypeLoadException e)
+                    {
+                        var sbMessage = new StringBuilder();
+                        sbMessage.AppendLine($"Error While loading {path}");
+                        sbMessage.AppendLine("-- Successfully Read Types --");
+                        foreach (var t in e.Types)
+                            if (t != null) sbMessage.AppendLine(t.ToString());
```

Comment:
Is printing all loaded types warranted? My main concern is that if the assembly has a dozen of them, printing all loaded types seems like it could bloat the console.

Reply:
I thought about it, and decided to follow.
Because I wanted to output much information as possible, I added `foreach (var t in e.Types)`. After all, I didn't use it. Because the elimination method was tough work.
I think We should avoid the risk of bloating the console.